### PR TITLE
Revisit s3_release_bucket usage in build scripts

### PR DESCRIPTION
**Pull request by pzbitskiy**
## Summary

Make S3_RELEASE_BUCKET specification explicit in build scripts.

### CODE REVIEW

File: scripts/deploy_version.sh [hunk 1]
```diff
@@ -23,6 +24,7 @@ set -e
 if [ "$#" -lt 2 ]; then
     echo "Syntax: deploy_version <branch> <os>/<arch> ..."
     echo "e.g. deploy_version master darwin/amd64 linux/amd64"
+    echo "Target S3 bucket must be set as S3_RELEASE_BUCKET env var"
```
Comment (tsachiherman) [inline scripts/deploy_version.sh]: You'll need to add explicit testing for this env variable in this script.

File: scripts/upload_version.sh [hunk 2]
```diff
@@ -12,18 +12,14 @@
 # Usage:    Should be used after build_packages.sh completes successfully
 #           Currently called by deploy_version.sh
 #
-# Examples: scripts/upload_version.sh nightly $HOME/node_pkg/
+# Examples: scripts/upload_version.sh nightly $HOME/node_pkg/ algorand-releases
 
 if [[ "$#" -ne 2 && "$#" -ne 3 ]]; then
-    echo "Syntax: upload_version <channel> <directory-with-packages> <optional: bucket>"
+    echo "Syntax: upload_version <channel> <directory-with-packages> <bucket>"
```
Comment (tsachiherman) [inline scripts/upload_version.sh]: The if clause above is no longer correct. the number of argument need to be exactly 3.

Comment (pzbitskiy) [reply scripts/upload_version.sh]: done

**Comment by pzbitskiy** [comment]
Thanks for catching this.

**Comment by winder** [comment]
This seems fine to me.

Could you update this document to reflect these changes:
https://algorand.atlassian.net/wiki/spaces/DEVOPS/pages/420118553/Usage+Scenarios

**Comment by pzbitskiy** [comment]
> Looks like `scripts/travis/deploy_packages.sh` needs to be updated as well.

@winder what exactly needs to be changed there? deploy_packages.sh is travis script that relies on environment.

**Comment by pzbitskiy** [comment]
> Could you update this document to reflect these changes:

Looks up to date.

**Comment by pzbitskiy** [comment]
> I think that we'll need a corresponding change in upload_config.sh.

Done

**Comment by pzbitskiy** [comment]
1. Checked that deploying private network works as before.
2. Fixed missed hexdump error in build in Docker (discovered when checked p.1)

**Pull request merged**